### PR TITLE
Adding another grok pattern to the filebeats mongo module pipeline.

### PR DESCRIPTION
**Pull request by cwray**
Adding another grok pattern to the filebeats mongo module, ingest pipeline.

We have some extended logging turned on in mongo. The log file format that mongo creates for these extended logs is not parseable by the current grok filter. Adding another grok statement to the list of Grok statements to handle these extended mongo logs.

see.
https://discuss.elastic.co/t/filebeat-mongo-module-grok-pattern-is-incorrect/138971/4

### CODE REVIEW

**Comment by cwray** [comment]
Added.

**Comment by ruflin** [comment]
@cwray Thanks for the contribution. Could you share a few log lines from a mongodb log file to test against the new pattern? 

I initially wanted to ask you to add it to our existing test file but just realised it's missing :-( Will try to open a PR to add one so you can add it then.

Could you add an entry to the changelog file?

**Comment by kvch** [comment]
Do you mind rebasing your PR on top of mine? This contains the missing example logs: https://github.com/elastic/beats/pull/7568

**Comment by cwray** [comment]
For the logs that don't match.

```
2018-07-09T17:05:25.294+0000 I -        [conn212435] end connection 127.0.0.1:43454 (479 connections now open)
2018-07-09T17:05:26.003+0000 I -        [conn212344]   Index Build (background): 18324200/222053333 8%
```

The log lines will work fine. 

```
2018-07-09T17:05:25.243+0000 I NETWORK  [thread1] connection accepted from 127.0.0.1:43454 #212435 (479 connections now open)
2018-07-09T17:05:25.266+0000 I ACCESS   [conn212435] Successfully authenticated as principal adcellerantClusterAdministrator on admin
2018-07-09T17:05:26.923+0000 I COMMAND  [conn212367] command adc command: aggregate ----REDACTED---- protocol:op_query 350ms
```

It might aslo be good to catch anyting with a COMMAND statement and parse them out diffrently To be able to look at what commands are running and how long they are taking. But that is for another day.

**Comment by James-Quigley** [comment]
FWIW, I was experiencing an issue related to this: https://discuss.elastic.co/t/mongodb-module-grok-expression-does-not-match/139241/4

However creating a pipeline manually with the grok patterns in this PR still did not fix my problem. The patterns pass when using the simulate pipeline, but not when using filebeat.

**Comment by cwray** [comment]
@James-Quigley

What does your pipeline look like?

Here is what is currently working for me.


```
"filebeat-6.3.0-mongodb-log-pipeline": {
    "on_failure": [
      {
        "set": {
          "field": "error.message",
          "value": "{{ _ingest.on_failure_message }}"
        }
      }
    ],
    "description": "Pipeline for parsing MongoDB logs",
    "processors": [
      {
        "grok": {
          "field": "message",
          "patterns": [
            "%{TIMESTAMP_ISO8601:mongodb.log.timestamp} %{WORD:mongodb.log.severity} %{WORD:mongodb.log.component} *\\[%{WORD:mongodb.log.context}\\] %{GREEDYDATA:mongodb.log.message}",
            "%{TIMESTAMP_ISO8601:mongodb.log.timestamp} %{WORD:mongodb.log.severity} - *\\[%{WORD:mongodb.log.context}\\] %{GREEDYDATA:mongodb.log.message}"
          ],
          "ignore_missing": true
        }
      }
    ]
  }
```

You will need to add this pipeline, Then remove the current index and restart filebeats. Im not sure why I had to remove the current index to get this all to work.

**Comment by James-Quigley** [comment]
@cwray I copied the default filebeat pipeline, but just changed the grok patterns. I've totally purged ES of indexes related to filebeat and I removed the existing pipelines before adding the custom pipeline.

When the documents get ingested into ES though, I get "Grok expressions do not match".

Just so I'm clear though, what do you mean when you say "Remove the current index"

**Comment by ruflin** [comment]
@cwray As https://github.com/elastic/beats/pull/7568 is merged, could you rebase you changes on top of master?

**Comment by cwray** [comment]
@ruflin I would love to but I'm a bit new to this process. What is the best way to handle the rebase and push back to this branch?

**Comment by kvch** [comment]
Add elastic/beats to your remotes, if you don't have it already.
```
git remote add upstream https://github.com/elastic/beats
git fetch upstream
```
Rebase your branch.
```
git checkout add_mongo_grok_pattern
git rebase upstream/master
```
Resolve conflicts.
```
git add filebeat/module/mongodb/log/ingest/pipeline.json
git rebase --continue
```
Or if you can't add your patch run `git rebase --skip`, to skip that patch.

Force push it to your repo.
```
git push origin add_mongo_grok_pattern -f
```

Also, it's a good idea to create a backup branch before start rebasing. If things get ugly, you can revert to your backup.
Let me know if you need further help.

**Comment by cwray** [comment]
Ok, I think I managed to unscrew my commit from yesterday and rebase. Let me know if I can do anything else.

**Comment by James-Quigley** [comment]
@cwray I'm still not able to get this working. I copied exactly what is in the commit and created a new pipeline. Deleted the indices I'm going to be writing to. Then restarted filebeat using the new pipeline and the grok expressions still don't match. 

Am I missing something obvious? The exact same pipeline passes when using the simulate API

File: filebeat/module/mongodb/log/ingest/pipeline.json [hunk 1]
```diff
@@ -4,7 +4,9 @@
     "grok": {
       "field": "message",
       "patterns":[
-        "%{TIMESTAMP_ISO8601:mongodb.log.timestamp} %{WORD:mongodb.log.severity} %{WORD:mongodb.log.component} \\s*\\[%{WORD:mongodb.log.context}\\] %{GREEDYDATA:mongodb.log.message}"
+
```
Comment (ruflin) [inline filebeat/module/mongodb/log/ingest/pipeline.json]: Can you remove this empty line?

Comment (cwray) [reply filebeat/module/mongodb/log/ingest/pipeline.json]: Sure I will give all this  build stuff a go if I can get a chance today.

Comment (kvch) [reply filebeat/module/mongodb/log/ingest/pipeline.json]: What is the error you are getting?
You need to put you example logs in `filebeat/module/mongodb/log/test/mongodb-debian-3.2.11.log`. Before running the generator command, you need to build filebeat using `make filebeat.test` and then run `GENERATE=1 INTEGRATION_TESTS=1 TESTING_FILEBEAT_MODULES=mongodb nosetests tests/system/test_modules.py`.

**Comment by James-Quigley** [comment]
FWIW I'm still having this issue. Can't get our mongo logs into ES because of it

**Comment by cwray** [comment]
Sorry, this has not been revisited in a while. I have been pulled off this project at my company and resigned to a different task.

**Comment by kaiyan-sheng** [comment]
This problem is already fixed with the latest filebeat mongodb grok pattern. 
With input log:
```
2018-07-10T20:42:37.863+0000 I NETWORK [thread1] connection accepted from 10.6.4.44:56076 #26676 (42 connections now open)
```

Output expected json is:
```
    {
        "@timestamp": "2018-07-10T20:42:37.863Z",
        "event.dataset": "mongodb.log",
        "event.module": "mongodb",
        "fileset.name": "log",
        "input.type": "log",
        "log.level": "I",
        "log.offset": 102,
        "message": "connection accepted from 10.6.4.44:56076 #26676 (42 connections now open)",
        "mongodb.log.component": "NETWORK",
        "mongodb.log.context": "thread1",
        "service.type": "mongodb"
    }
```

Will close this PR now.

**Pull request closed without merging**